### DISCUSSION TITLE
SASS e/ou PostCSS

### DISCUSSION THREAD

**Discussion by celsomtrindade** [frontendbr/forum#513] [Perguntas e Respostas]
Fala galera!
Recentemente aprofundei minhas leituras em `PostCss` e isso me abriu um mar de "dúvidas", pois eu migrei para o `SASS` justamente por algumas facilidades de desenvolvimento que o mesmo proporciona, por exemplo:

- Utilizar variáveis;
- Utilizar `@mixin` para reaproveitamento de código;
- Escrever `css` _nested_;
- Criação de algumas `functions`, como função para gerar _grid system_;

Então eu tenho uma estrutura de código mais ou menos assim:

- variáveis; //Todos os arquivos contendo variáveis, ex.: _margin, _index, _components, _colors, _typo
- functions; //Todas as declarações de funções e mixins
- ui; //Elementos UI comuns, ex.: form, button, etc...
- view; //Layout de views únicas, ex.: inicio, contato, etc...

Tudo isso eu faço utilizando `SASS` e, ao executar o `Grunt`, ele irá gerar um arquivo de @import com ordem baseado na ordem das pastas, depois concatenar todos os arquivos, gerando um único `css` compilado, convertendo variáveis, etc.

Porém, notei que o mesmo objetivo pode ser atingido utilizando apenas o `PostCSS` (e os respectivos _plugins_, lógico). Então, gostaria de saber qual a opinião de vocês a respeito disso, acham que devo migrar para o `PostCSS` ou continuar com esse método de `SASS`? Normalmente trabalho em projetos desde sites institucionais até projetos um tanto quanto extensos, contendo mais de 30-40 views, etc.., Sei que é algo meio "pessoal", mas como não utilizo `PostCSS`, gostaria de saber o outro lado da moeda (ou os dois lados, caso alguém já utilizou os 2).

O que consegui encontrar em leituras foi que, o `SASS` terá suporte para prefixos, automatizando um pouco mais o fluxo de compilação do `css` final, e no lado do `PostCSS`, a velocidade de compilação, visto que estamos trabalhando somente com `css` e alguns `plugins` para processamento final.

**Comment by nandomoreirame** [comment]
SASS ❤️

**Comment by felipefialho** [comment]
Linkando: #187 #31

**Comment by eduardojmatos** [comment]
@celsomtrindade estou usando o PostCSS com algumas ferramentas, como o Autoprefixer, CSS nano e mqpacker. Gosto bastante da abordagem de manter o Sass (por hora) e usar o PostCSS como ferramenta complementar. 

Dropar o Sass ainda é meio difícil, principalmente nos projetos que trabalho, visto que utilizamos vários mixins e functions. Se não tiver essa dependência, acho super justo usar "CSS vanilla" combinado com PostCSS.

**Comment by ghost** [comment]
@celsomtrindade Eu larguei o Sass e fui para o Stylus, até chegar no PostCSS. Tudo é muito parecido, mas há algumas diferenças qnd olhamos para a proposta de cada um, e também o gosto pessoal é algo que determina muito.

Atualmente montei o meu workflow ([Estag Starter](https://github.com/wdss93/estag-starter)) com poucos plugins para o PostCSS, peguei apenas os mais importantes com intuito de não deixá-lo muito grande.

**Comment by mksanti** [comment]
@celsomtrindade Faz um ano que troquei SASS pelo PostCSS. O motivo foi pela praticidade em ferramentas como Autoprefixer, Pixrem, Lost (sistema de grid) e para trazer o poder do SASS uso este cara daqui [`precss`](https://github.com/jonathantneal/precss)

p.s.[Uso um workflow parecido ao seu, se quiser dar uma olhada](https://github.com/mikejavier/devframework)

**Comment by felipefialho** [comment]
Sinceramente, prefiro utilizar Pré-processador + PostCSS.

Apesar de saber das vantagens de modularizar apenas o que realmente vai precisar através do PostCSS, mantenho um pré-processador (Stylus ❤️ ) no processo de build e complemento com features extras de PostCSS, como o Rucksack, Font Magician, Lost e CSSNano.

Funciona bem comigo.
